interfaces/acados_c/dense_qp.c

```diff
@@ -0,0 +1,257 @@
+/*
+ *    This file is part of acados.
+ *
+ *    acados is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation; either
+ *    version 3 of the License, or (at your option) any later version.
+ *
+ *    acados is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with acados; if not, write to the Free Software Foundation,
+ *    Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+
+#include "acados_c/dense_qp.h"
+
+//external
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+//acados
+#include <acados/dense_qp/dense_qp_common.h>
+#include <acados/dense_qp/dense_qp_hpipm.h>
+#include <acados/dense_qp/dense_qp_qore.h>
+#include <acados/dense_qp/dense_qp_qpoases.h>
+
+
+
+void dense_qp_copy_dims(dense_qp_dims *dest, dense_qp_dims *src)
+{
+	dest->nv = src->nv;
+	dest->ne = src->ne;
+	dest->nb = src->nb;
+	dest->ng = src->ng;
+	dest->ns = src->ns;
+	dest->memsize = src->memsize;
+}
+
+
+
+void dense_qp_copy_args(dense_qp_solver_plan *plan, dense_qp_dims *dims, void *dest, void *src)
+{
+    //TODO(nielsvd): remove the hack below. It breaks when the args used
+    //                         to construct the solver gets out of scope.
```

Comment:
All data is copied, but the pointers are not updated to the new location. It works, as long as the original ``args``-struct exists. Working on fixing this as we speak.

Reply:
I see, thanks for the clarification! So do you think we can ultimately remove solver-specific includes in the example files? E.g.:

https://github.com/acados/acados/blob/new_core/examples/c/mass_spring_example.c#L27
https://github.com/acados/acados/blob/new_core/examples/c/mass_spring_example.c#L31

we were wondering with @roversch whether this is possible and how. But that's not an urgent matter of course since C users are expected to have some more familiarity with the code.

Reply:
Ideally we get rid of solver-specific includes. So far I only needed them to cast arguments structs.

There's a draft prototype of an option-setting method [here](https://github.com/acados/acados/blob/new_core/interfaces/acados_c/options.h) to make this easier. My first idea is to copy ``casadi``'s as much as possible here, hence string-based and expressing hierarchy through periods.

Other ideas are welcome.